pkg/test/framework/components/authz/server.go

```diff
@@ -95,3 +123,37 @@ func SetupLocal(server *Server, ns namespace.Getter) resource.SetupFn {
 		return err
 	}
 }
+
+func AddPullSecret(deploymentYaml string, pullSecret string) (string, error) {
```

Comment:
Could we just move this to the `yml` package? Seems like it might be generally useful? It's conceptually similar to `yml.ApplyNamespace`.